Add private eventbrite logic

descholar-ceo opened the pull request:
## What does this PR do?
It adds the entire `private event-brite` logic

## Tasks to be completed in this PR
- [x] Project setup
- [x] Write the actual logic of the private Eventbrite
- [x] Add some stylings
- [x] Fix some linters error
- [x] Polish codes

## How to test this PR on your local machine?
1. Clone this repository with `git clone https://github.com/descholar-ceo/private-events.git`
1. Go inside this repo locally by typing `cd private-events`
1. Check out this branch by running `git checkout add-events-brite-logic`
1. Run `bundle install` to install all ruby gems
1. Run `yarn` to install Javascript dependencies
1. Run `bin/rails db:migrate` to create necessary tables in the database
1. Run `rails server`
1. Navigate to `http://127.0.0.1:3000` in your browser
1. Signup and create some events
1. To create an invitation click the `invite someone` in the navigation bar
1. To see a list of all events, click on `All events` button on the navbar
1. To see event's show page, navigate to the `all events` page and click on any event's `show` button

descholar-ceo committed:
Setup auth part

Georjane committed:
Installed bundle and yarn

Georjane committed:
Scaffold Invitation and Event models

Georjane committed:
Fix rubocop errors

Georjane committed:
Fix Stylelint error

descholar-ceo committed:
Setup the show action for users, and show user's events

descholar-ceo committed:
Fix the failed show of lists of events created with one user

Georjane committed:
Add view for new event

Georjane committed:
Add view for show event

Georjane committed:
Add view for index events

Georjane committed:
Fix lint errors

descholar-ceo committed:
Show the attended events on the users show page and event attendees on the event's show page

descholar-ceo committed:
Fix the rubocop lint error

Georjane committed:
Add scope for users and events

Georjane committed:
Prefered String literal

Georjane committed:
Create partial table for Event view

Georjane committed:
Create partial table for User view

Georjane committed:
Add User helper method

descholar-ceo committed:
Fix the rendering view error

descholar-ceo committed:
Remove the commented codes

descholar-ceo committed:
Fix rubocop lint error

Georjane committed:
Add application helper method

descholar-ceo committed:
Add some stylings

descholar-ceo committed:
Add template of the readme file

Georjane committed:
Update READMe

descholar-ceo committed:
Add screenshots to the readme file

descholar-ceo committed:
Add more description to the readme file

rOluochKe reviewed (changes requested):
Hello @descholar-ceo and @Georjane 
### You have done well so far :clap: However, you have a few issues to fix before the project can be approved:-
- [x] The event's show page displays the list of attendants
- [x] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming
### Make the changes and submit for another code review. Good luck.

By Raymond
Slack @iamraymond

descholar-ceo committed:
Implement feedback after the first code review

-Add more concise explanation how to run this project on local computer
-Add more navigation buttons in the navbar for the sake of the ease navigation through the project

descholar-ceo commented:
> Hello @descholar-ceo and @Georjane
> 
> ### You have done well so far  However, you have a few issues to fix before the project can be approved:-
> * [x]  The event's show page displays the list of attendants
> * [x]  The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming
> 
> ### Make the changes and submit for another code review. Good luck.
> By Raymond
> Slack @IAmRaymond

Hello @rOluochKe, Thank you for this valuable feedback which made us make our project docs a bit more concise. Actually the features you are talking about are already included in the project, as you can see them properly on the following screenshots:

`The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming`
![screencapture-localhost-3000-users-2-2020-09-04-13_33_41](https://user-images.githubusercontent.com/56886064/92235059-5de3fa80-eeb3-11ea-999c-3e00abc70dbb.png)

`The event's show page displays the list of attendants`
![screencapture-localhost-3000-events-4-2020-09-04-13_32_25](https://user-images.githubusercontent.com/56886064/92234956-32611000-eeb3-11ea-9361-0c1f6ca1d411.png)

 Maybe the problem is that is not well documented how to test it, We have edited our PR description to show how to test this PR. and edited our readme file as well, to explain concisely about running this project.

mosesogwo reviewed (approved):
**Hello! :wave:. Great job on this :+1: Project is approved.**

However, note that you are expected to write both unit and integration tests for this project. Sorry that it wasn't mentioned earlier.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png)

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png)

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png)

_**Do well to add these tests before merging the PR.**_

_**Good luck :stars:**_

### Status: Approved :heavy_check_mark:
